iai_donbot:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 125

  # Trajectory Controllers 
  whole_body_controller/base:
    type: effort_controllers/JointTrajectoryController
    joints:
    - odom_x_joint
    - odom_y_joint
    - odom_z_joint
    gains:
      odom_x_joint: {p: 100000.0, i: 10000, d: 100.0}
      odom_y_joint: {p: 100000.0, i: 10000, d: 100.0}
      odom_z_joint: {p: 100000.0, i: 10000, d: 100.0}
    constraints:
      goal_time: 0.6
      odom_x_joint: {trajectory: 0.2, goal: 0.1}
      odom_y_joint: {trajectory: 0.2, goal: 0.1}
      odom_z_joint: {trajectory: 0.2, goal: 0.1}
    stop_trajectory_duration: 0.5
    state_publish_rate: 125
    action_monitor_rate: 10

  whole_body_controller/body:
    type: effort_controllers/JointTrajectoryController
    joints:
    - ur5_shoulder_pan_joint
    - ur5_shoulder_lift_joint
    - ur5_elbow_joint
    - ur5_wrist_1_joint
    - ur5_wrist_2_joint
    - ur5_wrist_3_joint
    gains:
      ur5_shoulder_pan_joint: {p: 5000.0, i: 10, d: 100.0}
      ur5_shoulder_lift_joint: {p: 5000.0, i: 10, d: 100.0}
      ur5_elbow_joint: {p: 5000.0, i: 10, d: 100.0}
      ur5_wrist_1_joint: {p: 5000.0, i: 10, d: 10.0}
      ur5_wrist_2_joint: {p: 5000.0, i: 10, d: 10.0}
      ur5_wrist_3_joint: {p: 5000.0, i: 10, d: 10.0}
      gripper_base_gripper_left_joint: {p: 5000.0, i: 100, d: 1.0}
      gripper_base_gripper_right_joint: {p: 5000.0, i: 100, d: 1.0}
    constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      ur5_shoulder_pan_joint: {trajectory: 0.2, goal: 0.1}
      ur5_shoulder_lift_joint: {trajectory: 0.2, goal: 0.1}
      ur5_elbow_joint: {trajectory: 0.2, goal: 0.1}
      ur5_wrist_1_joint: {trajectory: 0.2, goal: 0.1}
      ur5_wrist_2_joint: {trajectory: 0.2, goal: 0.1}
      ur5_wrist_3_joint: {trajectory: 0.2, goal: 0.1}
    stop_trajectory_duration: 0.5
    state_publish_rate: 125
    action_monitor_rate: 10

  whole_body_controller/gripper:
    type: effort_controllers/JointTrajectoryController
    joints:
    - gripper_base_gripper_left_joint
    - gripper_base_gripper_right_joint
    gains:
      gripper_base_gripper_left_joint: {p: 500.0, i: 10, d: 0.1}
      gripper_base_gripper_right_joint: {p: 500.0, i: 10, d: 0.1}
    stop_trajectory_duration: 0.5
    state_publish_rate: 125
    action_monitor_rate: 10

  hardware_interface:
    sim_control_mode: 0
    joints:
      - odom_x_joint
      - odom_y_joint
      - odom_z_joint
      - ur5_shoulder_pan_joint
      - ur5_shoulder_lift_joint
      - ur5_elbow_joint
      - ur5_wrist_1_joint
      - ur5_wrist_2_joint
      - ur5_wrist_3_joint
      - gripper_base_gripper_left_joint
      - gripper_base_gripper_right_joint
  generic_hw_control_loop:
    loop_hz: &loop_rate 100
    cycle_time_error_threshold: 0.015
